.github/scripts/windows_install_utils.ps1

```diff
@@ -0,0 +1,105 @@
+# Taken from actions/virtual-environments
+function Install-Binary
+{
+    <#
+    .SYNOPSIS
+        A helper function to install executables.
+    .DESCRIPTION
+        Download and install .exe or .msi binaries from specified URL.
+    .PARAMETER Url
+        The URL from which the binary will be downloaded. Required parameter.
+    .PARAMETER Name
+        The Name with which binary will be downloaded. Required parameter.
+    .PARAMETER ArgumentList
+        The list of arguments that will be passed to the installer. Required for .exe binaries.
+    .EXAMPLE
+        Install-Binary -Url "https://go.microsoft.com/fwlink/p/?linkid=2083338" -Name "winsdksetup.exe" -ArgumentList ("/features", "+", "/quiet")
+    #>
+
+    Param
+    (
+        [Parameter(Mandatory)]
+        [String] $Url,
+        [Parameter(Mandatory)]
+        [String] $Name,
+        [String[]] $ArgumentList
+    )
+
+    Write-Host "Downloading $Name..."
+    $filePath = Start-DownloadWithRetry -Url $Url -Name $Name
+
+    # MSI binaries should be installed via msiexec.exe
+    $fileExtension = ([System.IO.Path]::GetExtension($Name)).Replace(".", "")
+    if ($fileExtension -eq "msi")
+    {
+        $ArgumentList = ('/i', $filePath, '/QN', '/norestart')
+        $filePath = "msiexec.exe"
+    }
+
+    try
+    {
+        Write-Host "Starting Install $Name..."
+        $process = Start-Process -FilePath $filePath -ArgumentList $ArgumentList -Wait -PassThru
+
+        $exitCode = $process.ExitCode
+        if ($exitCode -eq 0 -or $exitCode -eq 3010)
```

Comment:
Can you please add comment that `3010 == ERROR_SUCCESS_REBOOT_REQUIRED` (and perhaps a link to https://docs.microsoft.com/en-us/windows/win32/msi/error-codes

Reply:
+1 on links in comments in general